scripts/update/radarr.sh

```diff
@@ -1,77 +1,78 @@
-			echo_log_only "Radarr API tested and reachable"
-		fi
-		echo_progress_done
-		#
-		ret=$(curl -sL "http://127.0.0.1:7878/api/v3/system/status?apiKey=${apikey}")
-		#
-		echo_log_only "Content of ret =\n ${ret}"
-		if echo "$ret " | jq . >> "$log" 2>&1; then
-			isnetcore=$(jq '.isNetCore' <<< "$ret")
-		else
-			echo_log_only "jq decided ret wasn't valid"
-		fi
-
-		##TODO find a different way to check this seeing as we need to query Radarr API, would ben nicer to do from FS
-		if [[ $isnetcore = "false" ]]; then # This case confirms we are running on v3 without .net core, i.e. the case we want to update
-
-			echo_progress_start "Downloading source files"
-			if ! curl "https://radarr.servarr.com/v1/update/master/updatefile?os=linux&runtime=netcore&arch=x64" -L -o /tmp/Radarr.tar.gz >> "$log" 2>&1; then
-				echo_error "Download failed, exiting"
+. /etc/swizzin/sources/functions/radarr
+if [[ $(_radarr_version) != 'mono-v2' ]]; then
+	if [[ -f /install/.radarr.lock ]]; then
+		#Move v3mono installs to v3.net
+		if grep -q "ExecStart=/usr/bin/mono" /etc/systemd/system/radarr.service; then
+			echo_info "Moving Radarr from mono to .Net"
+			echo_log_only "Found radarr service pointing to mono"
+			#shellcheck source=sources/functions/utils
+			. /etc/swizzin/sources/functions/utils
+			[[ -z $radarrOwner ]] && radarrOwner=$(_get_master_username)
+			apikey=$(grep -oPm1 "(?<=<ApiKey>)[^<]+" /home/"${radarrOwner}"/.config/Radarr/config.xml)
+			echo_log_only "Apikey = $apikey"
+			#
+			echo_progress_start "Waiting for Radarr"
+			if ! timeout 30 bash -c -- "while ! curl -fL \"http://127.0.0.1:7878/api/v3/system/status?apiKey=${apikey}\" >> \"$log\" 2>&1; do sleep 5; done"; then
+				echo_warn "Radarr API did not respond as expected. Please make sure Radarr is on v3 and running."
 				exit 1
+			else
+				urlbase="$(curl -sL "http://127.0.0.1:7878/api/v3/config/host?apikey=${apikey}" | jq '.urlBase' | cut -d '"' -f 2)"
+				echo_log_only "Radarr API tested and reachable"
+			fi
+			echo_progress_done
+			#
+			ret=$(curl -sL "http://127.0.0.1:7878/api/v3/system/status?apiKey=${apikey}")
+			#
+			echo_log_only "Content of ret =\n ${ret}"
+			if echo "$ret " | jq . >> "$log" 2>&1; then
+				isnetcore=$(jq '.isNetCore' <<< "$ret")
+			else
+				echo_log_only "jq decided ret wasn't valid"
 			fi
-			echo_progress_done "Source downloaded"
 
-			echo_progress_start "Extracting archive"
-			systemctl stop radarr -q
-			rm /opt/Radarr/Radarr.exe
-			tar -xvf /tmp/Radarr.tar.gz -C /opt >> "$log" 2>&1
-			chown -R "$radarrOwner":"$radarrOwner" /opt/Radarr
-			echo_progress_done "Archive extracted"
+			##TODO find a different way to check this seeing as we need to query Radarr API, would ben nicer to do from FS
+			if [[ $isnetcore = "false" ]]; then # This case confirms we are running on v3 without .net core, i.e. the case we want to update
```

Comment:
Is there a better way to determine if we're on mono now, other than querying the API?

Reply:
we can use the function, if that is all we need to know?